examples/sinh_extension/setup.py

```diff
@@ -23,15 +26,55 @@
     "include",
 )
 
-nvfuser_lib_dir = str(
-    pathlib.Path(importlib.util.find_spec("nvfuser").origin).parent / "lib"
-)
+# Ensure nvfuser_common is installed before trying to find its path
+try:
+    nvfuser_common_spec = importlib.util.find_spec("nvfuser_common")
+    if nvfuser_common_spec is None or nvfuser_common_spec.origin is None:
+        raise ImportError("Could not find nvfuser_common. Is it installed?")
+    nvfuser_lib_dir = str(pathlib.Path(nvfuser_common_spec.origin).parent / "lib")
+except ImportError as e:
+    print(f"Error finding nvfuser_common path: {e}")
+    print("Ensure 'nvfuser_common' is installed in the build environment.")
+    raise e
+
+
+# Inherit from PyTorch BuildExtension
+# Modify build_extension from setuptools.command.build_ext to move shared
+# library to nvfuser_extension package
+class custom_build_ext(BuildExtension):
+    def build_extension(self, ext):
+        # Handle different os and cpu arch
+        def _find_library_path():
+            for item in os.listdir("build"):
+                if item.startswith("lib"):
+                    return item
+            raise RuntimeException("Cannot find lib in build directory")
+
+        # Call PyTorch BuildExtension first that overloads
+        # setuptools.command.build_ext
+        super().build_extension(ext)
+        if ext.name == "nvfuser_extension":
+            # Copy files on necessity.
+            filename = self.get_ext_filename(self.get_ext_fullname(ext.name))
+            fileext = os.path.splitext(filename)[1]
+
+            extension_path = os.path.join("./build/", _find_library_path(), filename)
+            assert os.path.exists(extension_path)
+
+            install_dst = os.path.join("nvfuser_extension", filename)
+            if not os.path.exists(os.path.dirname(install_dst)):
+                os.makedirs(os.path.dirname(install_dst))
+            self.copy_file(extension_path, install_dst)
```

Comment:
@jjsjann123 I feel like this should be done automatically but we do something similar for nvfuser.

Reply:
I think we did the copy via `package_data` in setup function.

Reply:
But given that this is just an example, I'm less worried about it.